src/main/java/com/direwolf20/buildinggadgets/common/building/modes/ExchangingSurfaceMode.java

```diff
@@ -0,0 +1,46 @@
+package com.direwolf20.buildinggadgets.common.building.modes;
+
+import com.direwolf20.buildinggadgets.common.BuildingGadgets;
+import com.direwolf20.buildinggadgets.common.building.IPlacementSequence;
+import com.direwolf20.buildinggadgets.common.building.IValidatorFactory;
+import com.direwolf20.buildinggadgets.common.building.Region;
+import com.direwolf20.buildinggadgets.common.building.placement.ConnectedSurface;
+import com.direwolf20.buildinggadgets.common.building.placement.Surface;
+import com.direwolf20.buildinggadgets.common.building.placement.Wall;
+import com.direwolf20.buildinggadgets.common.items.gadgets.GadgetGeneric;
+import com.direwolf20.buildinggadgets.common.tools.GadgetUtils;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.math.BlockPos;
+
+/**
+ * Reference region and searching region are the same.
+ *
+ * @see BuildingSurfaceMode
+ */
+public class ExchangingSurfaceMode extends AbstractMode {
```

Comment:
?!? I have no idea what this description would like to tell me.
Please add some description of what it does.